### PR TITLE
JPERF-791 bump libraries to resolve lockfiles conflicts in ssh library

### PR DESCRIPTION
**Pull request by mgrzaslewicz**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -28,14 +28,14 @@ configurations.all {
 
 dependencies {
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
-    api("org.testcontainers:testcontainers:1.15.1")
+    api("org.testcontainers:testcontainers:1.17.3")
```
Comment (dagguh) [inline build.gradle.kts:31]: Is `testcontainers:1.17.3` incompatible with `testcontainers:1.15.1` and that's why we need to change `SshUbuntuContainer`?

Comment (dagguh) [reply build.gradle.kts:31]: Ah, no, the rationale is in the commit message :+1: 

> To resolve conflicting dependency net.java.dev.jna:jna: 5.2.0 vs 5.5.0
in ssh library which is consumer of this library

So even if jna:5.5.0 is compatible with jna:5.2.0, then we're getting bogus conflicts. But which lib is declaring compatibility with jna:5.2.0, while not declaring it with jna:5.5.0?

Comment (mgrzaslewicz) [reply build.gradle.kts:31]: It seems com.github.docker-java:docker-java-transport-zerodep in ssh-ubuntu.
ssh-ubuntu hasn't been updated in ssh library for a long time, there is 0.1.0 version

```
❯ cd ssh-ubuntu
❯ ./gradlew -q dependencies
+--- org.testcontainers:testcontainers:1.15.1
|    +--- junit:junit:4.12
|    |    \--- org.hamcrest:hamcrest-core:1.3
|    +--- org.slf4j:slf4j-api:1.7.30 -> 1.7.25
|    +--- org.apache.commons:commons-compress:1.20
|    +--- org.rnorth.duct-tape:duct-tape:1.0.8
|    +--- org.rnorth.visible-assertions:visible-assertions:2.1.2
|    |    \--- net.java.dev.jna:jna:5.2.0 -> 5.5.0
|    +--- com.github.docker-java:docker-java-api:3.2.7
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.3
|    |    \--- org.slf4j:slf4j-api:1.7.30 -> 1.7.25
|    \--- com.github.docker-java:docker-java-transport-zerodep:3.2.7
|         +--- com.github.docker-java:docker-java-transport:3.2.7
|         +--- org.slf4j:slf4j-api:1.7.25
|         \--- net.java.dev.jna:jna:5.5.0
```

Comment (dagguh) [reply build.gradle.kts:31]:
```
|    +--- org.rnorth.visible-assertions:visible-assertions:2.1.2
|    |    \--- net.java.dev.jna:jna:5.2.0 -> 5.5.0
```
They are probably compatible with 5.5.0 as well, but too lazy to express that via dependency range declarations.

So this PR is trying to find a matching permutation of versions, which do not suffer from such a bogus conflict. It's a lot of work on our (consumer) side, while the lib producer is the cause. We already paid the development price here, so we can merge, but for the future we can use other, cheaper or maintainable solutions:
* remove dependency on a lazy lib
* fix the lib declaration:
  * report it as a bug/issue
  * OR contribute the fix
* manually ignore/resolve the bogus conflict via `useVersion` in `build.gradle.kts` (we chose this one most often)

Comment (dagguh) [reply build.gradle.kts:31]: PS. `docker-java-transport-zerodep` promised zero dependencies, has 3, what a ripoff :wink:

**Comment by dagguh** [comment]
It changes `src/main` and important parts of the pom.xml, so let's explain the change in `CHANGELOG.md`. Consumers need a reason to bump.

**Review by dagguh — CHANGES_REQUESTED** [review]
It changes `src/main` and important parts of the pom.xml, so let's explain the change in `CHANGELOG.md`. Consumers need a reason to bump.

**Comment by mgrzaslewicz** [comment]
@dagguh I'm not sure if I know what you mean

> It changes src/main and important parts of the pom.xml, so let's explain the change in CHANGELOG.md. Consumers need a reason to bump.

The only difference in pom.xml is testcontainers 1.15.1 -> 1.17.3 and it's already in the changelog

**Comment by dagguh** [comment]
I don't see any diff in `CHANGELOG.md` and POM change is not the only one. What do the changes in `src/main` bring to the consumers?

**Comment by mgrzaslewicz** [comment]
@dagguh oh, I have modified that locally only, forgot to commit and push

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,10 @@ Dropping a requirement of a major version of a dependency is a new contract.
 ## [Unreleased]
 [Unreleased]: https://github.com/atlassian/ssh-ubuntu/compare/release-0.2.5...master
 
+### Changed
+- Bump testcontainers library to 1.17.3
+- Generate SshUbuntuContainer in a way it needs no modifications of container after starting it. This way it should be less sensitive to changes in behaviour of feature testcontainer library changes.
```
Comment (dagguh) [inline CHANGELOG.md:28]: Awesome :heart:

**Review by dagguh — APPROVED** [review]

**Pull request merged**